distribution/integration/tests-integration/tests/src/test/java/org/wso2/carbon/integration/tests/featuremanagement/FeatureInstallationTestCase.java

```diff
@@ -64,11 +62,15 @@ public void testGracefulServerRestart() throws Exception {
         featureManager.reviewInstallFeatures();
         featureManager.getLicensingInformation();
         featureManager.installFeatures();
+        log.info("Feature :" + featureList.get(0).getFeatureID() + " installed successfully");
         serverAdminClient = new ServerAdminClient(automationContext);
+        log.info("Going to restart the server");
         serverAdminClient.restartGracefully();
-        ClientConnectionUtil.waitForPort(Integer.parseInt(automationContext.getDefaultInstance()
-                                                                  .getPorts().get("https")), 240000
-                , true, "localhost");
+        Thread.sleep(10000);
```

Comment:
Method serverAdminClient.restartGracefully() is asynchronous after calling that it take sometime to down the server. Since waitForPort method calls before it shutdown it returns success(It thinks server has restarted becuase port is open) and proceed but at that time only server is shitting down